docs/tutorials/task.csv-to-oscal-cd/csv-to-oscal-cd.md

```diff
@@ -0,0 +1,151 @@
+# task csv-to-oscal-cd
+
+The below table represents the expectations of trestle task `csv-to-oscal-cd` for the contents of the input csv-file for synthesis of the output OSCAL Component Definition json-file.
+
+`Column Name` is the name of the expected column in the input csv-file. Any additional columns not identified here, for example foobar, are also extracted and placed into the output json-file as component.control-implementation.prop\["foobar"\].
+
+`Component Definition Locale` is the `path` within the output json-file into witch the value is stashed.
+
+An example input [csv-file](ocp4-sample-input.csv).
+
+An example output [json-file](component-definition.json).
+
+<table>
+
+<tr style="text-align:left;vertical-align:top">
+<th>Column Name
+<th>Value Type
+<th>Specification
+<th>Value Description
+<th>Component Definition Locale
+<th>Example Value
+
+<tr style="text-align:left;vertical-align:top">
+<td>Rule_Id
+<td>String
+<td>required
+<td>A textual label that uniquely identifies a policy (desired state) that can be used to reference it elsewhere in this or other documents.
+<td>component.control-implementation.prop["Rule_Id"]
+<td>password_policy_min_length_characters
+
+<tr style="text-align:left;vertical-align:top">
+<td>Rule_Description
+<td>String
+<td>required
+<td>A description of the policy (desired state) including information about its purpose and scope.
+<td>component.control-implementation.prop["Rule_Description"]
+<td>Ensure password policy requires minimum length of 12 characters
+
+<tr style="text-align:left;vertical-align:top">
+<td>Profile_Reference_URL
+<td>String
+<td>required
+<td>A URL reference to the source catalog or profile for which this component is implementing controls for. A profile designates a selection and configuration of controls from one or more catalogs
+<td>component.control-implementation.source
+<td>https://github.com/usnistgov/oscal-content/blob/main/nist.gov/SP800-53/rev5/json/NIST_SP-800-53_rev5_HIGH-baseline_profile.json
+
+<tr style="text-align:left;vertical-align:top">
+<td>Profile_Description
+<td>String
+<td>required
+<td>A description of the profile.
+<td>component.control-implementation.description
+<td>NIST Special Publication 800-53 Revision 5 HIGH IMPACT BASELINE
+
+<tr style="text-align:left;vertical-align:top">
+<td>Component_Type
+<td>String
+<td>required
+<td>A category describing the purpose of the component.
+<td>component.type
+<td>Validation
```

Comment:
"describing the purpose ..." its not purpose but type of a component like :service", "software", etc. The types are pre-defined by OSCAL so one of those types should be used.